src/bootstrap/test.rs

```diff
@@ -1365,11 +1365,14 @@ note: if you're sure you want to do this, please open an issue as to why. In the
         }
 
         let mut flags = if is_rustdoc { Vec::new() } else { vec!["-Crpath".to_string()] };
-        if !is_rustdoc {
+        if !is_rustdoc && mode != "ui" {
             if builder.config.rust_optimize_tests {
```

Comment:
Can we remove this block entirely so the logic can live fully in compiletest rather than being split between there and bootstrap?

Reply:
I'll give it a try